WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsListFragment.java

```diff
@@ -131,13 +154,21 @@ public void removeComment(Comment comment) {
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
+
         setUpListView();
         getCommentAdapter().loadComments();
+
+        Bundle extras = getActivity().getIntent().getExtras();
+        mHasAutoRefreshedComments = extras.getBoolean(CommentsActivity.KEY_AUTO_REFRESHED);
+        mEmptyViewMessageType = MessageType.getEnumFromString(extras.getString(CommentsActivity.KEY_EMPTY_VIEW_MESSAGE));
+
         if (!NetworkUtils.isNetworkAvailable(getActivity())) {
+            updateEmptyView(MessageType.NETWORK_ERROR);
```

Comment:
This works as expected when switching from another page to the Comments page, displaying Network Error. There's an edge case when the user switches blogs where it will display 'Comments couldn't be refreshed at this time' when it should be Network Error. I can post a video if this needs clarification.

Look into the UpdateCommentsTask, onPostExecute is toasting and displaying a GENERIC_ERROR because it doesn't check for a network connection.